### ISSUE TITLE
Unexpected PR list from release-notes 0.3.3

### ISSUE THREAD

**Issue by joejulian** [kubernetes/release#1368]
#### What happened:
The testing branch between tags testing-1.7.0 and testing-1.8.0 had only 1 PR with release notes. The tool created 7 entries with PRs that were included with the previous release.

#### What you expected to happen:
I expected only PR #249 to be included.

#### How to reproduce it (as minimally and precisely as possible):
release-notes --start-sha $(git rev-parse "testing-1.7.0") --end-sha $(git rev-parse "testing-1.8.0") --github-repo kubernetes-base-addons --github-org mesosphere --required-author "" --format json --output /tmp/rn3 --dependencies=false
--branch=testing

#### Environment:

- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Others:

**Comment by puerco** [comment]
Thanks for the report @joejulian I'm on it
/assign

**Comment by puerco** [comment]
Hey @joejulian I cloned the repository and ran `release-notes` on it with the same params and got the same results. After reviewing the output I found that the document has the correct notes included.

I think you are not considering several Pull Requests that fall into the range you're specifying in the above invocation or the release-notes tool.

You can see all the commits and PRs that are analyzed by the release notes tool using those ranges here:

https://github.com/mesosphere/kubernetes-base-addons/commits/testing?after=10d8ad3ed5ef62a7a23e98b457ef4a8e9d60d46d+0

(the first commit on that page `5d79c2a` corresponds to the `--start-sha` and the `--end-sha` which is `aa256d` is almost at the end of the page on Apr 23).

I went deeper and the notes included in the output match those PRs in the range which have a release note defined:

| PR | Release Note |
| --- | --- |
| mesosphere/kubernetes-base-addons#193 | [dex-k8s-authenticator] added support for the konvoy credentials plugin |
| mesosphere/kubernetes-base-addons#196 | [cert-manager]: `usages` is no longer definable as part of `issuerRef`, instead it is a key on its own |
| mesosphere/kubernetes-base-addons#212 | [dex-k8s-authenticator]: Now supports a kubectl credentials plugin for automatically  |managing identity tokens. Instructions for downloading the plugin and configuring kubectl can be found at `https://<cluster-ip>/token/plugin`.  |
| mesosphere/kubernetes-base-addons#230 | No |
| mesosphere/kubernetes-base-addons#244 | Kubernetes upgrade to 1.17.x ... |
| mesosphere/kubernetes-base-addons#246 | UI Bump 3.46.5 -> 3.58.0 |
| mesosphere/kubernetes-base-addons#249 | \[kibana\]: Fixes an issue causing an outdated version of Kibana to be deployed to GCP. |

I count ten other PRs without a release note which are correctly excluded.

I hope that seeing the commits and PRs this way helps to get a clearer view of how the changes in the source are taken into account when generating the release notes document.

**Comment by joejulian** [comment]
That entire list, everything < 249 was in the prior tag. Note that PR 193 is commit e6eb3c8faf2a099a4257bb541f66a5f7da96727c.
```
$ git tag --contains e6eb3c8faf2a099a4257bb541f66a5f7da96727c
stable-1.15-1.7.0
stable-1.15-1.8.0
stable-1.16-1.7.0
stable-1.16-1.8.0
stable-1.17-1.8.0
testing-1.16-2.0.1
testing-1.17-2.0.1
testing-1.7.0
testing-1.8.0
testing-1.9.0
testing-2.0.0
testing-2.0.0-3
testing-2.0.1
testing-2.0.2
```
Since we're only asking for commits after `testing-1.7.0` up to `testing-1.8.0` I would expect that any commits that are included in 1.7.0 would not be also included in 1.8.0.

**Comment by joejulian** [comment]
Looks like this is a problem with my git tree and process. Closing.
